lib/config.js

```diff
@@ -66,13 +66,14 @@ class Config {
         this.parser = options.parser;
         this.parserOptions = options.parserOptions || {};
 
+        this.configCache = new ConfigCache();
```

Comment:
Also, it's worth noting that the flattening behavior has been present since 4.0.0, so it's not part of this regression.